Mathlib/Topology/EMetricSpace/Defs.lean

```diff
@@ -538,8 +534,17 @@ end Compact
 
 end EMetric
 
---namespace
-/-- We now define `EMetricSpace`, extending `PseudoEMetricSpace`. -/
+/-- An extended metric space is a T1 topological space and uniform space equipped with an extended
+metric `edist`, possibly taking the value `∞`, such that the topology comes from metric.
+
+We make the uniformity/topology part of the data instead of deriving it from the metric. This eg
+ensures that we do not get a diamond when doing
+`[EMetricSpace α] [EMetricSpace β] : TopologicalSpace (α × β)`: The product metric and product
+topology agree, but not definitionally so.
+
+When instantiating `EMetricSpace`, the uniformity fields are not necessary, they
+will be filled in by default. There is a default value for the uniformity, that can be substituted
+in cases of interest, for instance when giving an `EMetricSpace` instance on a product. -/
```

Comment:
I don't think this is helpful here: at this point there's not even an expectation that the uniformity fields would be necessary!

Reply:
Hmm... that part I took from the existing docstrings